### ISSUE TITLE
@spider=fedex has point without main tag for convenience shop providing package services

### ISSUE THREAD

**Issue by matkoniecz** [alltheplaces/alltheplaces#5881]
Ideally it would be marked that it is not separate POI, but extra property of some existing POI to be attached to. In such case many of tags listed here should be dropped. (or should it be separate POI? in such case main tag is missing, not entirely sure about tagging)

https://www.alltheplaces.xyz/map/#16.82/50.072221/20.041573

![screen01](https://github.com/alltheplaces/alltheplaces/assets/899988/46772969-5a13-4a19-9b09-3eb5bde7a36f)

**Comment by panasonic-mafia** [comment]
I believe it should be a separate POI. Category tag was missing due to failed NSI match because [Fedex NSI](https://nsi.guide/index.html?t=brands&tt=fedex) has multiple categories.

**Comment by matkoniecz** [comment]
If that applies `amenity=post_office` tag then it is still wrong. For reference, right now many small shops in Poland allow picking up packages from one or more delivery services. Adding `amenity=post_office` near them is definitely not a valid tagging.

**Comment by panasonic-mafia** [comment]
> For reference, right now many small shops in Poland allow picking up packages from one or more delivery services.

Ah I see what you mean. 

Ref: 

- https://local.fedex.com/en-pl/warsaw/waw6e
- and POI data for above POI at https://validator.schema.org/#url=https%3A%2F%2Flocal.fedex.com%2Fen-pl%2Fwarsaw%2Fwaw6e

The name of such POIs has 'OnSite' ([more info](https://www.fedex.com/en-pr/shipping/onsite.html)) which may help to identify them from standalone locations. I'm not sure about tagging though. Perhaps `description` tag can be added, something like e.g. `<service name> inside <name of OnSite location>`

**Comment by Cj-Malone** [comment]
In OSM this is probably `post_office=post_partner` as an attribute on another POI. https://wiki.openstreetmap.org/wiki/Key:post_office for these locations.

We can't merge this into an attribute of another POI in ATP. Anyone consuming this data would have to.

**Comment by Cj-Malone** [comment]
> rapid is showing this data to import

Not really, Rapid can show pmtiles now, in the same way JOSM can open GeoJSON.

Personally I hope ATP to becomes a supported dataset, but I envision that as a separate server project. Where we would have rules like you are talking about. I'd exclude any POI without a top level category. I'd exclude spiders like visa, western union and gbfs. I'd exclude loads of tags that don't make sense in an OSM context.

**Comment by matkoniecz** [comment]
huh, this fake Fedex POI disappeared in Poland.

Has data source changed, spider failed, Fdex run away from Poland or what? Maybe no changes are needed?

Found some new improvements to data in Poland while scrolling but no Fedexes.

Looking at https://www.alltheplaces.xyz/spiders - Fedex spider died.

Is it #6433 or other issue? To make it more confusing it is possible to download geojson file with 22 764 lines, looking fine at first look but QGIS fails to open it.

**Comment by Cj-Malone** [comment]
If a spider hard crashes, the end of the json isn't written. You probably can edit, remove the last `,` and add `]}`.

What I haven't seen before is the stats failing (probably causing the 0 in the list). Something really crashed hard.

**Comment by iandees** [comment]
It could be that the spider timed out. I use the `timeout` command to force a scrapy run to fail when it takes too long (see [here](https://github.com/alltheplaces/alltheplaces/blob/master/ci/run_all_spiders.sh#L37)). The scrapy timeout configuration doesn't seem to work when a connection is held open on the other end. This seems to happen sometimes as a way to slow/stop spiders (or randomly due to networking problems).
